moderntribe/square-one
# Feature/phpcs workflow updates

## What does this do/fix?

- Currently if you have the `sniff` job from the PHPCS workflow set under "Require status checks to pass before merging" and no PHP files changed in our specified paths, the PR will not pass those checks and the PR can only be merged by administrators. 
- Uses [dorny/paths-filter ](https://github.com/dorny/paths-filter) action to detect file changes instead so the workflow will still pass if no PHP file changes are present in the PR.

## QA

**Test with no PHP file changes**
- Ensure the `sniff` job is added to "Require status checks to pass before merging"  under Settings > Branches > main > Protect matching branches.
- Change only a JS file and open a PR.
- The PHPCS workflow will now properly pass.

**Test PHP file changes in specified paths**
- Ensure the `sniff` job is added to "Require status checks to pass before merging"  under Settings > Branches > main > Protect matching branches
- Make a breaking PHPCS change to a PHP file under `wp-content/plugins/core/`, and/or `wp-content/themes/core/` and/or `wp-content/mu-plugins` and the job should properly fail.

## Tests

Does this have tests?

- [x] No, this doesn't need tests because it's a GitHub workflow change
